test/dds/core/xtypes/myTest.cpp

```diff
@@ -558,46 +565,780 @@ TEST (DynamicData, cascade_construction)
+    EXPECT_EQ(TypeConsistency::IGNORE_TYPE_SIGN, another_str.is_compatible(the_str));
+    EXPECT_EQ(TypeConsistency::IGNORE_TYPE_SIGN, another_str.is_compatible(other_str));
+}
+
+TEST (DynamicType, testing_is_compatible_structure_of_primitive_type_mixed_int)
+{
+    StructType the_str("check");
+    the_str.add_member(Member("int", primitive_type<uint16_t>()));
+    StructType other_str("other_check");
+    other_str.add_member(Member("int", primitive_type<uint32_t>()));
+    StructType another_str("another_check");
+    another_str.add_member(Member("int", primitive_type<int64_t>()));
+    EXPECT_EQ(TypeConsistency::IGNORE_TYPE_WIDTH, the_str.is_compatible(other_str));
+    EXPECT_EQ(TypeConsistency::IGNORE_TYPE_WIDTH |
+              TypeConsistency::IGNORE_TYPE_SIGN, the_str.is_compatible(another_str));
+    EXPECT_EQ(TypeConsistency::IGNORE_TYPE_WIDTH |
+              TypeConsistency::IGNORE_TYPE_SIGN, other_str.is_compatible(another_str));
+}
+
+TEST (DynamicType, testing_is_compatible_structure_of_array_same_bound)
+{
+    StructType the_str("check");
+    ArrayType the_array(primitive_type<uint32_t>(), 10);
+    the_str.add_member(Member("arr", the_array));
+    StructType other_str("other_check");
+    other_str.add_member(Member("arr", the_array));
+    EXPECT_EQ(TypeConsistency::EQUALS , the_str.is_compatible(other_str));
+    EXPECT_EQ(TypeConsistency::EQUALS , other_str.is_compatible(the_str));
+}
+
+TEST (DynamicType, testing_is_compatible_structure_of_array_different_bound_and_type)
+{
+    StructType the_str("check");
+    ArrayType the_array(primitive_type<uint32_t>(), 10);
+    the_str.add_member(Member("arr", the_array));
+
+    ArrayType other_array(primitive_type<uint32_t>(), 11);
+    StructType other_str("other_check");
+    other_str.add_member(Member("arr", other_array));
+
+    ArrayType another_array(primitive_type<int32_t>(), 10);
+    StructType another_str("other_check");
+    another_str.add_member(Member("arr", another_array));
+
+    EXPECT_EQ(TypeConsistency::IGNORE_ARRAY_BOUNDS , the_str.is_compatible(other_str));
+    EXPECT_EQ(TypeConsistency::IGNORE_TYPE_SIGN, the_str.is_compatible(another_str));
+    EXPECT_EQ(TypeConsistency::IGNORE_ARRAY_BOUNDS, other_str.is_compatible(the_str));
+    EXPECT_EQ(TypeConsistency::IGNORE_ARRAY_BOUNDS | TypeConsistency::IGNORE_TYPE_SIGN, other_str.is_compatible(another_str));
+    EXPECT_EQ(TypeConsistency::IGNORE_TYPE_SIGN, another_str.is_compatible(the_str));
+    EXPECT_EQ(TypeConsistency::IGNORE_ARRAY_BOUNDS | TypeConsistency::IGNORE_TYPE_SIGN, another_str.is_compatible(other_str));
+}
+
+TEST (DynamicData, testing_equality_check_primitive_type)
+{
+    {
+        DynamicData dd1(primitive_type<uint8_t>());
+        DynamicData dd2(primitive_type<uint8_t>());
+        dd1.value(uint8_t(15));
+        dd2.value(uint8_t(15));
+        EXPECT_EQ(dd1, dd2);
+        dd2.value(uint8_t(16));
+        EXPECT_NE(dd1, dd2);
+    }
```

Comment:
Should something like the following pass the test? I think so.
```cpp
    {
        DynamicData dd1(primitive_type<uint32_t>());
        DynamicData dd2(primitive_type<int32_t>());
        dd1.value(uint32_t(15));
        dd2.value(int32_t(15));
        EXPECT_EQ(dd1, dd2);
        dd2.value(int32_t(16));
        EXPECT_NE(dd1, dd2);
    }
```
As the stored values are the same. The same logic should apply with each compatible primitive types.